vnext/Shared/WinRTWebSocketResource.cpp

```diff
@@ -123,13 +117,14 @@ WinRTWebSocketResource::~WinRTWebSocketResource() noexcept /*override*/
 
 #pragma region Private members
 
-IAsyncAction WinRTWebSocketResource::PerformConnect() noexcept {
+IAsyncAction WinRTWebSocketResource::PerformConnect(Uri &&uri) noexcept {
   auto self = shared_from_this();
+  auto coUri = std::move(uri);
```

Comment:
>coUri [](http://example.com/codeflow?start=7&length=5)

Why is this needed? Why not just do std::move on line 127?

Reply:
Oh, because we want to std::move before co_await?

Reply:
But still, why do we need that here? wouldn't "uri" be still available after co_await?

Reply:
Nope. Unless it's a thread-safe pass by reference (it's not), a new lvalue must be held for the co-routine to keep it when it resumes.

Reply:
See https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/concurrency#parameter-passing